drivers/ethernet/eth_nxp_enet_qos/eth_nxp_enet_qos_mac.c

```diff
@@ -226,25 +237,39 @@ static void eth_nxp_enet_qos_rx(struct k_work *work)
 		if (net_recv_data(data->iface, pkt)) {
 			LOG_ERR("RECV failed");
 			/* Quite a shame. */
-			goto error;
+			/* error during processing, we continue with new allocated */
+			net_pkt_unref(pkt);
+			eth_stats_update_errors_rx(data->iface);
 		}
 
-		LOG_DBG("Recycling RX buf");
-
+		LOG_DBG("Swap RX buf");
 		/* Fresh meat */
 		data->rx.reserved_bufs[i] = new_buf;
 		desc->read.buf1_addr = (uint32_t)new_buf->data;
-		desc->read.control |= rx_desc_refresh_flags;
+		desc->read.control = rx_desc_refresh_flags;
 
 		/* Record our glorious victory */
 		eth_stats_update_pkts_rx(data->iface);
 	}
 
-	return;
+	/* try to restart if halted */
+	const struct device *dev = net_if_get_device(data->iface);
+	const struct nxp_enet_qos_mac_config *config = dev->config;
+	enet_qos_t *const base = config->base;
+	const uint32_t dma_ch0_stat = base->DMA_CH[0].DMA_CHX_STAT;
 
-error:
-	net_pkt_unref(pkt);
-	eth_stats_update_errors_rx(data->iface);
+	if (ENET_QOS_REG_GET(DMA_CH_DMA_CHX_STAT, RBU, dma_ch0_stat)) {
+		LOG_DBG("handle RECEIVE BUFFER UNDERRUN in worker");
+		base->DMA_CH[0].DMA_CHX_STAT = ENET_QOS_REG_PREP(DMA_CH_DMA_CHX_STAT, RBU, 0b1);
+
+		/* When the DMA reaches the tail pointer, it suspends. Set to last descriptor */
+		base->DMA_CH[0].DMA_CHX_RXDESC_TAIL_PTR = ENET_QOS_REG_PREP(
```

Comment:
This could be possibly be just rewritten to the same value it already has:

base->DMA_CH[0].DMA_CHX_RXDESC_TAIL_PTR = base->DMA_CH[0].DMA_CHX_RXDESC_TAIL_PTR;

I don't know what is better.